Version v1.0.1

cosminbosutar opened the pull request:
Redesigned the configuration page to make it more user friendly.

cosminbosutar committed:
Updated the admin module (plugin) configuration page to match the one from woocommerce.

cosminbosutar committed:
Removed the "active" select from the configuration view.

cosminbosutar committed:
Updated the package.xml file with the new version details.

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,346 @@
+            $liveMode = Mage::getStoreConfig('payment/tpay/liveMode', $storeId);
 
-            $data['checksum'] = $checksum;
+            Mage::log('CASDASDASD', null, 'twispay.log', true);
```
Please remove the line `Mage::log('CASDASDASD', null, 'twispay.log', true);`

cosminbosutar replied:
Removed.

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,346 @@
+            // @codingStandardsIgnoreStart
+            $iv = base64_decode ( $encryptedParts [0] );
+            if (false === $iv) {
+                Mage::log ( "Invalid encryption iv", Zend_Log::DEBUG, $this->logFileName );
```
The throw exception code lines from the `decryptResponse` method where removed but the calling code is testing for the exception ...

Also, this method will return multiple types now: NULL, FALSE or the decrypted string. 

(please edit the method description, return type - as it is now - is not `array`)

cosminbosutar replied:
I replaced the function with a new one in the package helper.

dragos-ursu commented on app/code/community/Twispay/Tpay/Model/PaymentMethod.php:
```diff
@@ -1,64 +1,69 @@
-        return $redirect_url;      
+    public function getOrderPlaceRedirectUrl() 
+    {
+        Mage::log('getOrderPlaceRedirectUrl', null, 'twispay.log', true);
```
Please remove the line `Mage::log('getOrderPlaceRedirectUrl', null, 'twispay.log', true);`

cosminbosutar replied:
Removed

dragos-ursu commented on app/code/community/Twispay/Tpay/Model/PaymentMethod.php:
```diff
@@ -1,64 +1,69 @@
+     */
+    public function authorize(Varien_Object $payment, $amount) 
+    {
+        Mage::log('authorize', null, 'twispay.log', true);
```
Please remove the line `Mage::log('authorize', null, 'twispay.log', true);`

cosminbosutar replied:
Removed

dragos-ursu commented on app/code/community/Twispay/Tpay/Model/PaymentMethod.php:
```diff
@@ -1,64 +1,69 @@
+    public function capture(Varien_Object $payment, $amount) 
     {
-        Mage::Log('Step 1 Process: Create and capture the process', Zend_Log::DEBUG, $this->LOG_FILE_NAME);
+        Mage::log('capture', null, 'twispay.log', true);
```
Please remove the line `Mage::log('capture', null, 'twispay.log', true);`

cosminbosutar replied:
Removed

dragos-ursu commented on app/code/community/Twispay/Tpay/Model/PaymentMethod.php:
```diff
@@ -1,64 +1,69 @@
     {
-    	Mage::Log("running serverurl", Zend_Log::DEBUG, $this->LOG_FILE_NAME);
-    	return "testurlserver";
+        Mage::log('serverurl', null, 'twispay.log', true);
```
Please remove the line `Mage::log('serverurl', null, 'twispay.log', true);`

cosminbosutar replied:
Removed

dragos-ursu reviewed (changes requested):
Please check the comments.

cosminbosutar committed:
Updated the checksum calculation algorithm.
Added logic to process all the Twispay response states.
Moved helper code to to the module default helper.
Added debug and error logging. Error login will always be printed and debug logs are available only when logging is enabled.

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,280 @@
-            return (bool) $val;
+      /* Check if shiping price needs to be added. */
+      if(0 < $order->getShippingAmount()){
+        $index = count($items);
```
The value of `$index` is not used anywhere (line 97 override any previous values)

cosminbosutar replied:
I removed the line.

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,280 @@
+
+      $link = $url;
+      $payment = $order->getPayment();
+      $payment->setTransactionId(time()); // Make it unique.
```
There could be two (or more) requests processed in the same second by the server, using just `time()` call does not ensure uniqueness. Previous version of code also appended `$orderId` ...

cosminbosutar replied:
I can append or prepend the $orderId.

Can you the format I want to use?

The previous version used only the time (please see https://github.com/Twispay/magento1.x/blob/master/app/code/community/Twispay/Tpay/controllers/PaymentController.php line 64 + line 127).
Maybe you are referring to the "identifier" field that is set to the Twispay server which was calculated like this ($data['identifier'] = "_".$rmTranid;) and is now calculated like following:
'identifier' => (0 == $billingAddress->getCustomerId()) ? ('_' . $orderId . '_' . date('YmdHis')) : ('_' . $billingAddress->getCustomerId())

dragos-ursu replied:
Yes, sorry my mistake. The value of `identifier` it's ok. If the value of `$payment->setTransactionId(...)` need to be unique, we should use something more than just calling `time()`, prepending `orderId` should do it.

cosminbosutar replied:
I prepended the $orderId. The transaction ID is now constructed like following:
 $payment->setTransactionId($orderId . '_' . time());

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,280 @@
+
+    /* Extract the order. */
+    $orderId = explode('_', $decrypted['externalOrderId'])[0];
+    $order = Mage::getModel('sales/order')->loadByIncrementId($orderId);
```
Variable `$order` is not used, is there any reason to call `loadByIncrementId`?

cosminbosutar replied:
I removed the line.

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,280 @@
+
+    /* Extract the order. */
+    $orderId = explode('_', $decrypted['externalOrderId'])[0];
+    $order = Mage::getModel('sales/order')->loadByIncrementId($orderId);
```
Variable `$order` is not used, is there any reason to call `loadByIncrementId`?

cosminbosutar replied:
I removed the line.

dragos-ursu reviewed (changes requested):
Please see the comments

cosminbosutar committed:
Removed lines after review.

cosminbosutar committed:
Prepende the orderId to the transactionId to make it unique.

cosminbosutar committed:
Added debug and error logging. Error login will always be printed and debug logs are available only when logging is enabled.&#xD;
Added contact email field to the configuration page.&#xD;
Added checkout success and failure pages.&#xD;
Added en_US lang file.

cosminbosutar committed:
Added methods and config option for authorize and capture payment steps.

cosminbosutar committed:
Added functionality to create a transaction after each successful payment.

cosminbosutar committed:
Removed config select that allowed the user to select bwtween authorization or authorization and capture to always force the second option.

cosminbosutar committed:
indentation

cosminbosutar committed:
Added support for refund

cosminbosutar committed:
Partial refund bugfix and moved logged messages to the lang file.

cosminbosutar committed:
Temp commit

cosminbosutar committed:
Created separate backUrl and IPN sratus update functions for purchase and recurring profile orders.
Created functions for extracting config values.
Added limitations info in plugin config page.
Added observer for "checkoutSubmitAllFunction" that is used for setting the redirect URL for paying a recurring profile.

cosminbosutar committed:
Addd purchase order refund and partial refund suport.

cosminbosutar committed:
Added purchase order type refund and partial refund support.
Added recurring order type refund and partial refund support.
Added daily sync with the Twispay server for recurring profiles.

dragos-ursu commented on app/code/community/Twispay/Tpay/controllers/PaymentController.php:
```diff
@@ -1,362 +1,563 @@
+            }
+
+          } catch (Exception $e) {
+            Mage::logException($exception);
```
Please check this line, `$exception` is undefined at this point.

cosminbosutar replied:
Added the correct value.

dragos-ursu commented on app/code/community/Twispay/Tpay/Helper/Data.php:
```diff
@@ -1,8 +1,1097 @@
+                            , Mage_Sales_Model_Order::STATE_CANCELED
+                            , Mage::helper('tpay')->__(' Order #%s canceled as payment for transaction #%s has been charged back.', $purchase->getIncrementId(), $transactionId));
+
+        Mage::Log(__FUNCTION__ . Mage::helper('tpay')->__(' [RESPONSE]: Status charge-back for order ID: ') . $profile->getId(), Zend_Log::NOTICE , $this->logFileName);
```
Please check this line, `$profile` is undefined at this point.

cosminbosutar replied:
Added the correct value.

dragos-ursu commented on app/code/community/Twispay/Tpay/Helper/Data.php:
```diff
@@ -1,8 +1,1097 @@
+    $response = json_decode($response);
+
+    /* Check if the response code is 200 and message is 'Success'. */
+    if ((200 == $response->code) && ('Success' == $response->message)) {
```
We should also check if the response was decoded successfully with `json_decode`.

cosminbosutar replied:
Added the check.

dragos-ursu commented on app/code/community/Twispay/Tpay/Helper/Data.php:
```diff
@@ -1,8 +1,1097 @@
+    $response = json_decode($response);
+
+    /* Check if the response code is 200 and message is 'Success'. */
+    if ((200 == $response->code) && ('Success' == $response->message)) {
```
We should also check if the response was decoded successfully with `json_decode`.

cosminbosutar replied:
Added the check.

dragos-ursu commented on app/code/community/Twispay/Tpay/Helper/Data.php:
```diff
@@ -1,8 +1,1097 @@
+    }
+
+    /* JSON decode the decrypted data. */
+    return json_decode($decryptedResponse, /*assoc*/TRUE, /*depth*/4);
```
We should also check if the response was decoded successfully with `json_decode`, otherwise return false.

cosminbosutar replied:
Added the check.

dragos-ursu commented on app/code/community/Twispay/Tpay/Model/PaymentMethod.php:
```diff
@@ -1,64 +1,222 @@
+    $response = json_decode($response);
+
+    /* Check if the response code is 200 and message is 'Success'. */
+    if ((200 == $response->code) && ('Success' == $response->message)) {
```
We should also check if the response was decoded successfully with `json_decode`.

cosminbosutar replied:
Added the check.

dragos-ursu reviewed (changes requested):
Please check the comments

cosminbosutar committed:
Changes after review.

dragos-ursu commented:
code review done
